sysutils/afscompress/Portfile

```diff
@@ -0,0 +1,63 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem           1.0
+PortGroup            github 1.0
+
+github.setup         RJVB afsctool 1.7.0
+
+name                 afscompress
+categories           sysutils
+platforms            darwin
+license              GPL-3
+maintainers          {@ylluminarious orbitalimpact.com:georgedp} openmaintainer
+description          A fork of brkirch's afsctool utility, featuring several improvements.
+long_description     AFSC (Apple File System Compression) tool is a utility that can be used to \
+    apply HFS+/APFS compression to file(s), decompress HFS+/APFS compressed file(s), \
+    or get information about existing HFS+/APFS compressed file(s). macOS 10.6 or \
+    later is required. This fork has several modifications, mostly concerning the \
+    compression feature, including: improved error reporting, an attempt to reduce \
+    memory pressure pressure compressing large files, support for multiple \
+    files/folders specified on the commandline, a backup option while compressing \
+    (that comes in addition to the existing undo if something went wrong), and \
+    support for files that are read-only (and/or write-only) by changing their \
+    permissions temporarily. No error checking is done for this feature\; failure \
+    will lead to errors that are already caught. The main new feature that justifies \
+    the version bump, however, is the parallel processing feature, allowing the user \
+    to specify an arbitrary (though positive :)) number of threads that will \
+    compress the specified files in parallel.
+
+checksums            sha256  911ae33960560e36d854753cca1fcb83bb4674123b83f04b5642980670bc747d \
+                     rmd160  b2ec4858164723532bee7d3b00f796d42aefbaea \
+                     size    110026
+
+depends_lib          port:sparsehash \
+                     port:zlib
+
+depends_build        port:cmake \
+                     port:pkgconfig
+
+use_configure        no
+universal_variant    yes
+
+pre-fetch {
+    if {${os.platform} eq "darwin" && ${os.major} < 10} {
+        ui_error "${name} ${version} is only compatible with Mac OS X 10.6 or later"
+        return -code error "incompatible Mac OS X version"
+    }
+}
+
+build {
+    system -W ${worksrcpath}       "mkdir ./build"
+    if {[variant_isset original_name]} {
+        system -W ${worksrcpath}/build "cmake -DCMAKE_INSTALL_PREFIX=${destroot}/${prefix} -Wno-dev .."
+    } else {
+        system -W ${worksrcpath}/build "cmake -DNEW_DRIVER_NAMES=ON -DCMAKE_INSTALL_PREFIX=${destroot}/${prefix} -Wno-dev .."
+    }
+    system -W ${worksrcpath}/build "make"
+}
```

Comment:
Use the cmake portgroup instead.

Reply:
@mf2k How do I pass in the correct build options with the `cmake` group? Also, do I have to replace the `github` group with `cmake`? If so, then I would have to do a lot of rigmarole to get it fetching the tarball from Github properly. Sadly, there's no documentation on the `cmake` group [in the MacPorts guide.](https://guide.macports.org/chunked/reference.portgroup.html)

Reply:
You can use both portgroups. Best to look at other existing ports for guidance. Look at `vigra` for instance.